### PR TITLE
Undo the TLS flags extension for negotiating RRC

### LINKED ISSUE #6: Can we use TLS flags?
We'd like to use [TLS Flags](https://datatracker.ietf.org/doc/html/draft-ietf-tls-tlsflags) to negotiate RRC.

At the moment, the document seems to be defined for (D)TLS 1.3 only, but the reason why the same scheme would not be applicable to 1.2 are not fully clear.

For now (-01), optimistically, we are using it in a version agnostic fashion, hoping that eventually that will be the case.

Achim has opened an [issue on the TLS flags repo](https://github.com/tlswg/tls-flags/issues/13).  This issue is for tracking on the RRC side.

### PR DESCRIPTION
**Pull request by thomas-fossati**
Use a new extension type instead.

Fixes #6

### CODE REVIEW

**Comment by boaks** [comment]
Yes, if tls-flags are not usable for DTLS 1.2, then go with this.

File: draft-ietf-tls-dtls-rrc.md [hunk 1]
```diff
@@ -268,21 +271,19 @@ harm to connectivity.
...
 
-- Message: CH,SH
+IANA is requested to allocate the extension code point (TBD1) for the `rrc`
+extension to the `TLS ExtensionType Values` registry as described in
+{{tbl-ext}}.
 
-- Recommended: Y
-
-- Reference: [[This document]]
+| Value | Extension Name | TLS 1.3 |  DTLS-Only |  Recommended | Reference |
```
Comment (thomas-fossati) [inline draft-ietf-tls-dtls-rrc.md:283]:
```suggestion
| Value | Extension Name | TLS 1.3 | DTLS-Only | Recommended | Reference |
```

File: draft-ietf-tls-dtls-rrc.md [hunk 2]
```diff
@@ -89,12 +89,15 @@ described in Section 4 of {{!RFC8446}}.
 
 # RRC Extension
 
-This specification uses the tls_flags extension defined in {{!I-D.ietf-tls-tlsflags}}
-to allow a client and a server to negotiate support for this extension.
-
-The RRC flag is assigned the value (TBD1) and is used in the ClientHello (CH) and
-the ServerHello (SH).
+The use of RRC is negotiated via the `rrc` DTLS-only extension.  On connecting,
+the client includes the `rrc` extension in its Client Hello if it wishes to use
```
Comment (hannestschofenig) [inline draft-ietf-tls-dtls-rrc.md:93]:
```suggestion
the client includes the `rrc` extension in its ClientHello if it wishes to use
```

File: draft-ietf-tls-dtls-rrc.md [hunk 3]
```diff
@@ -89,12 +89,15 @@ described in Section 4 of {{!RFC8446}}.
...
+RRC.  If the server is capable of meeting this requirement, it responds with a
+`rrc` extension in its Server Hello.  The `extension_type` value for this
```
Comment (hannestschofenig) [inline draft-ietf-tls-dtls-rrc.md:95]:
```suggestion
`rrc` extension in its ServerHello.  The `extension_type` value for this
```

**Pull request merged**